### PR TITLE
Add support for tracing redis's 'batch' & 'multi'

### PR DESCRIPTION
**Pull request by pbadenski**

### CODE REVIEW

File: packages/zipkin-instrumentation-redis/test/integrationTest.js [hunk 1]
```diff
@@ -73,6 +73,65 @@ describe('redis interceptor', () => {
...
+    const tracer = new Tracer({ctxImpl, recorder});
+
+    const redis = getRedis(tracer);
+    redis.on('error', done);
+    tracer.setId(tracer.createRootId());
+    const ctx = ctxImpl.getContext();
+    redis.set('ping', 'pong', 10, () => {
+      ctxImpl.letContext(ctx, () => {
+        redis.batch([ ['get', 'ping'] ]).exec(() => {
+          const annotations = recorder.record.args.map(args => args[0]);
```
Comment (adriancole) [inline packages/zipkin-instrumentation-redis/test/integrationTest.js]: probably want to verify the span name here. Not sure, but possibly some way to encode the list of "multi-ops" into a tag aka binary annotation?

Comment (pbadenski) [inline packages/zipkin-instrumentation-redis/test/integrationTest.js]: Both make sense :) Have a look at the new commit, let me know if you like it!

File: packages/zipkin-instrumentation-redis/src/zipkinClient.js [hunk 2]
```diff
@@ -42,24 +43,44 @@ module.exports = function zipkinClient(
...
-        id = tracer.createChildId();
-        tracer.setId(id);
-        commonAnnotations(method);
-      });
-      const wrapper = mkZipkinCallback(callback, id);
-      const newArgs = [...args, wrapper];
-      actualFn.apply(this, newArgs);
-    };
-  });
+  const wrap = function(client, parentId) {
```
Comment (muymoo) [inline packages/zipkin-instrumentation-redis/src/zipkinClient.js]: Is `parentId` actually the parent in the trace or is it the shared trace id across multi/batch calls?

File: packages/zipkin-instrumentation-redis/test/integrationTest.js [hunk 3]
```diff
@@ -73,6 +79,53 @@ describe('redis interceptor', () => {
     });
   });
 
+  it('should add zipkin annotations for batch method', (done) => {
```
Comment (muymoo) [inline packages/zipkin-instrumentation-redis/test/integrationTest.js]: Is there a test for `multi`? The easiest thing to mess up is the variable scoping of nested functions and wrapping. I'd make couple solid unit tests that verify you can share a the same trace id across multiple nested wrapped command calls. The wrapping and `.apply` can be tricky.

File: packages/zipkin-instrumentation-redis/src/zipkinClient.js (hunk 2 above)
Comment (pbadenski) [reply packages/zipkin-instrumentation-redis/src/zipkinClient.js]: you are right, this is traceId, I fixed it in the latest commit

File: packages/zipkin-instrumentation-redis/test/integrationTest.js (hunk 3 above)
Comment (pbadenski) [reply packages/zipkin-instrumentation-redis/test/integrationTest.js]: I added one (I hope pretty comprehensive test) in cad0528

**Comment by pbadenski** [comment]
Added changes suggested by @muymoo

**Comment by adriancole** [comment]
terrible network here.. can someone take a look at appveyor and confirm/deny whether it is related?

**Comment by adriancole** [comment]
thx!

**Pull request merged**